version: '3.8'

services:
  app:
    container_name: urbanlegend_app
    build:
      context: .
      dockerfile: Dockerfile
    volumes:
      - .:/var/www/html
    networks:
      - urban_legend
    depends_on:
      - mysql

  nginx:
    image: nginx:latest
    container_name: urbanlegend_nginx
    ports:
      - "8000:80"  # Externo:Interno
    volumes:
      - .:/var/www/html
      - ./nginx/default.conf:/etc/nginx/conf.d/default.conf
    depends_on:
      - app
    networks:
      - urban_legend

  mysql:
    container_name: urbanlegend_mysql
    image: mysql:8.0
    ports:
      - '3306:3306'
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: urbanlegend
      MYSQL_PASSWORD: root
    volumes:
      - mysql_data:/var/lib/mysql
    networks:
      - urban_legend
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-proot"]
      retries: 3
      timeout: 5s

  phpmyadmin:
    container_name: urbanlegend_phpmyadmin
    image: phpmyadmin:latest
    ports:
      - '8080:80'
    environment:
      - PMA_HOST=mysql
      - MYSQL_ROOT_PASSWORD=root
    networks:
      - urban_legend

networks:
  urban_legend:
    driver: bridge

volumes:
  mysql_data:
    driver: local
